staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -758,13 +768,22 @@ func (s *store) ttlOpts(ctx context.Context, ttl int64) ([]clientv3.OpOption, er
 	if ttl == 0 {
 		return nil, nil
 	}
-	// TODO: one lease per ttl key is expensive. Based on current use case, we can have a long window to
-	// put keys within into same lease. We shall benchmark this and optimize the performance.
-	lcr, err := s.client.Lease.Grant(ctx, ttl)
-	if err != nil {
-		return nil, err
+	s.leaseLock.Lock()
+	defer s.leaseLock.Unlock()
+	// If the previous lease grants valid TTL, reuse it; Otherwise create
+	// a new one with a little extra time to make it possible to be reused.
+	now := time.Now().Unix()
+	if now+ttl <= s.prevLeaseExpirationTime {
+		// do nothing and reuse the previous lease
+	} else {
+		s.prevLeaseExpirationTime = now + leaseReuseLength + ttl
```

Comment:
Why not 
```
prevLeaseExpirationTime = now + leaseReuseLength
```

and check

```
if now > prevLeaseExpirationTime
```

Reply:
I guess the values of ttls might be different for different events?